### PR TITLE
feat: add loading animations and skeletons

### PR DESCRIPTION
**Pull request by hetfly**

### CODE REVIEW

File: src/components/Spinner/index.tsx [hunk 1]
```diff
@@ -12,9 +13,13 @@ type SpinnerProps = {
   centered?: boolean;
 };
 
-const Spinner: FC<SpinnerProps> = ({ size = 'lg', className, centered = false, ...rest }) => (
+const Spinner: FC<SpinnerProps> = ({ size = 'sm', className, centered = false, ...rest }) => (
```
Comment (octocat512) [inline src/components/Spinner/index.tsx:16]: Is it possible to make a generalized component? The big spinner and small spinner images are different? I'd like to use the same image. @hetfly Please get help from @parrot9design if the images are different.

File: src/pages/StatsPage/index.tsx [hunk 2]
```diff
@@ -159,7 +160,11 @@ export function StatsPage() {
               </StatsItemIcon>
               <StatsItemData>
                 <Typography fontWeight="normal" variant="h3">
-                  {overviewInfo.BALNMarketCap ? getFormattedNumber(overviewInfo.BALNMarketCap, 'currency0') : '-'}
+                  {overviewInfo.BALNMarketCap ? (
+                    getFormattedNumber(overviewInfo.BALNMarketCap, 'currency0')
+                  ) : (
+                    <LoaderComponent />
```
Comment (octocat512) [inline src/pages/StatsPage/index.tsx:166]: Currently, there are two placeholder animations. LoaderComponent and Skeleton. @parrot9design 

image
image

What about using the same animation?

File: src/sections/PairSection.tsx [hunk 3]
```diff
@@ -120,6 +158,39 @@ export default function PairSection() {
   );
 }
 
+const SkeletonPariPlaceholder = () => {
```
Comment (octocat512) [inline src/sections/PairSection.tsx:161]: For the code convention, Please move utility components, constant variables, and utility functions ahead of the main component.

File: src/sections/TokenSection.tsx [hunk 4]
```diff
@@ -117,3 +147,44 @@ function CurrencyIcon({ currencyKey }: { currencyKey: CurrencyKey }) {
 
   return <Icon width={40} height={40} />;
 }
+
+const SkeletonTokenPlaceholder = () => {
```
Comment (octocat512) [inline src/sections/TokenSection.tsx:151]: please move the code before the main component.

**Review by octocat512 — CHANGES_REQUESTED** [review]

File: src/sections/TokenSection.tsx (hunk 4 above)
Comment (hetfly) [reply src/sections/TokenSection.tsx]: done

File: src/sections/PairSection.tsx (hunk 3 above)
Comment (hetfly) [reply src/sections/PairSection.tsx]: done

File: src/pages/StatsPage/index.tsx (hunk 2 above)
Comment (hetfly) [reply src/pages/StatsPage/index.tsx:168]: Yea, I'm not sure about this as well. I changed the first one on the top for comparison. Let Lisa and Peter decide.

File: src/components/Spinner/index.tsx (hunk 1 above)
Comment (octocat512) [reply src/components/Spinner/index.tsx]: thanks

File: src/pages/StatsPage/index.tsx (hunk 2 above)
Comment (parrot9design) [reply src/pages/StatsPage/index.tsx:168]: Thanks for the comparison! Happy to use two different animations here, so you can change the `Total value locked` animation back to LoaderComponent. 😊

So we can maintain a consistent animation style going forwards, what are your thoughts on this as a general rule?

- Use LoaderComponent for single values, like the data points in the overview, collateral/loan, and governance sections on the Stats page.
- Use Skeleton for content that takes up more space/has multiple data points, like the Tokens and Exchange sections on the Stats page, and the content on the Vote page.

(@hetfly if we go by this logic, we should add the Skeleton animation to the Holdings section of the [Performance details](https://stats.balanced.network/performance-details) page, as well.)

Comment (hetfly) [reply src/pages/StatsPage/index.tsx:166]: Agreed, code updated

**Review by parrot9design — APPROVED** [review]

**Pull request merged**